esx_service/utils/kvESX.py

```diff
@@ -166,35 +168,44 @@ def delete(volpath):
     return True
 
 
+# Align a given string to the specified block boundary.
+def align_str(kv_str, block):
+   # Align string to the next block boundary. The -1 is to accommodate
+   # a newline at the end of the string.
+   aligned_len = int((len(kv_str) + block - 1) / block) * block - 1
+   return '{:<{width}}\n'.format(kv_str, width=aligned_len)
+
+
 # Load and return dictionary from the sidecar
 def load(volpath):
     metaFile = lib.DiskLib_SidecarMakeFileName(volpath, DVOL_KEY)
 
     try:
-        fh = open(metaFile, "r+")
-    except IOError:
-        logging.exception("Failed to open %s", metaFile);
+       with open(metaFile, "r") as fh:
+          kv_str = fh.read()
+    except:
+        logging.exception("Failed to access %s", metaFile);
         return None
 
-    kvDict = json.load(fh)
-
-    fh.close()
-
-    return kvDict
+    try:
+       return json.loads(kv_str)
+    except ValueError:
+       logging.exception("Failed to decode meta-data for %s", volpath);
+       return None
```

Comment:
nit: in both 'returns' in this function, just 'return' would be better (to indicate no return value is expected)